File: kotlin/series/src/main/kotlin/Series.kt
package Series

fun slices(sliceLength: Int, s: String): List<List<Int>> {
  require(s.length >= sliceLength) { "Slice length is too large" }
  require(sliceLength >= 1) { "Slice length is too small" }
  
  var res = arrayListOf<List<Int>>()
  var intList = s.toCharArray().map({ it.toInt() - '0'.toInt() })

  for (i in 0..(intList.size - sliceLength)) {
    res.add(intList.slice(IntRange(i, i + sliceLength - 1)))
  }

  return res
}

File: swift/series/Sources/Series/Series.swift
struct Series {
    let series: [Int]
    
    init(_ series: String) {
        self.series = series.map({ Int(String($0))! })
    }
    
    func slices(_ length: Int) -> [[Int]] {
        length > series.count ? [[Int]]() :
            (0...series.count - length).map { Array(series[$0..<($0 + length)]) }
    }
}
